### PR TITLE
fix(core): run init hooks in injection context 🌶️

### PR DESCRIPTION
**Pull request by eneajaho**
Just to start a discussion

### CODE REVIEW

**Comment by samuelfernandez** [comment]
This would be super 🌶!

File: packages/core/src/render3/hooks.ts [hunk 1]
```diff
@@ -8,19 +8,18 @@
 
 import {setActiveConsumer} from '@angular/core/primitives/signals';
 
+import {runInInjectionContext} from '../di';
```
Comment (JeanMeche) [inline packages/core/src/render3/hooks.ts:11]: Could you try to import from `'../di/contextual'` to break the circularity this introduces  ?

**Comment by Segdo** [comment]
What's the status of this pr? Would love to see this!

**Comment by eneajaho** [comment]
This got so many reactions which tells me that the feature is desired by the community 😄

Closing as too spicy 🌶️ 

(we can always open it back 🤌)

**Comment by mikelgo** [comment]
We need this! + functional hooks (like e.g. afterNextRender but for ngOnInit etc.)

**Comment by eneajaho** [comment]
I just want **DestroyRef** in ngOnInit that's it ☹️

**Pull request closed without merging**